impl/src/main/java/org/ehcache/config/builders/CacheManagerBuilder.java

```diff
@@ -203,16 +206,7 @@ T cast(EhcacheManager ehcacheManager) {
    * @return a new builder with the added default copier
    */
   public <C> CacheManagerBuilder<T> withCopier(Class<C> clazz, Class<? extends Copier<C>> copier) {
-    DefaultCopyProviderConfiguration service = configBuilder.findServiceByClass(DefaultCopyProviderConfiguration.class);
-    if (service == null) {
-      service = new DefaultCopyProviderConfiguration();
-      service.addCopierFor(clazz, copier);
-      return new CacheManagerBuilder<>(this, configBuilder.addService(service));
-    } else {
-      DefaultCopyProviderConfiguration newConfig = new DefaultCopyProviderConfiguration(service);
-      newConfig.addCopierFor(clazz, copier, true);
-      return new CacheManagerBuilder<>(this, configBuilder.removeService(service).addService(newConfig));
-    }
+    return ensureThenUpdate(DefaultCopyProviderConfiguration::new, existing -> existing.addCopierFor(clazz, copier, true));
```

Comment:
The behavior of the `with<X>` methods in `CacheManagerBuilder` hasn't been changed.  I'm tempted to leave the existing documentation as is and to add a sentence like I used in the new configuration builder methods:
> This will remove any existing configurations that are incompatible with the supplied one.

Thoughts?

Reply:
I've reviewed and refined all of these. We need to have a discussion about terminology for collection stuff though, since "Set" there makes little sense.